### PR TITLE
[1LP][WIPTEST] Added tenant parent-child quota validation tests

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
Added parametrized validation tests to check setting of parent-child tenant quota
{{ pytest: -vvvv cfme/tests/configure/test_access_control.py -k test_child_tenant_quota_validation }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 37689ee93aa1137ea29ab56d1811a7b6e4a119ae:

`cfme/tests/configure/test_access_control.py`:
- :bangbang: Line 537:13: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 970:101: **E501** *line too long (106 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3a07b45259c2fd770de9543df8e0f4000a30d5ed:

`cfme/tests/configure/test_access_control.py`:
- :bangbang: Line 537:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 540:13: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 969:63: **E251** *unexpected spaces around keyword / parameter equals*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by vatsalparekh** [comment]
59z would fail due to [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1511853)

File: cfme/tests/configure/test_access_control.py [hunk 1]
```diff
@@ -962,3 +960,44 @@ def test_tenant_quota_input_validate(appliance):
         view.form.fill({'{}_cb'.format(field[0]): True, '{}_txt'.format(field[0]): field[1]})
         assert view.save_button.disabled
         view.form.fill({'{}_cb'.format(field[0]): False})
+
+
+@pytest.fixture(scope='module')
+def first_level_tenant(appliance):
+    root_tenant = appliance.collections.tenants.get_root_tenant()
+    tenant = appliance.collections.tenants.create(name='tenant1_' + fauxfactory.gen_alphanumeric(),
```
Comment (quarckster) [inline cfme/tests/configure/test_access_control.py]: **REQUIRED**
Please use `format`

File: cfme/tests/configure/test_access_control.py [hunk 2]
```diff
@@ -962,3 +960,44 @@ def test_tenant_quota_input_validate(appliance):
...
+                                                  description=fauxfactory.gen_alphanumeric(),
+                                                  parent=root_tenant)
+    yield tenant
+    if tenant.exists:
+        tenant.delete()
+
+
+@pytest.fixture(params=[True, False], ids=['tenant-project', 'tenant-tenant'])
```
Comment (quarckster) [inline cfme/tests/configure/test_access_control.py]: **REQUIRED**
```python
@pytest.fixture(params=["projects", "tenants"], ids=['tenant-project', 'tenant-tenant'])
def collections(appliance, request):
    yield (appliance.collections.tenants, getattr(appliance.collections, request.param))

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/configure/test_access_control.py (hunk 1 above)
Comment (vatsalparekh) [reply cfme/tests/configure/test_access_control.py]: Done

File: cfme/tests/configure/test_access_control.py (hunk 2 above)
Comment (vatsalparekh) [reply cfme/tests/configure/test_access_control.py]: Done

File: cfme/tests/configure/test_access_control.py [hunk 3]
```diff
@@ -1001,3 +1001,41 @@ def test_tenant_quota_input_validate(appliance):
         view.form.fill({'{}_cb'.format(field[0]): True, '{}_txt'.format(field[0]): field[1]})
         assert view.save_button.disabled
         view.form.fill({'{}_cb'.format(field[0]): False})
+
+
+@pytest.fixture(scope='module')
+def first_level_tenant(appliance):
+    root_tenant = appliance.collections.tenants.get_root_tenant()
+    tenant = appliance.collections.tenants.create(name='Tenant_{}' + fauxfactory.gen_alphanumeric(),
```
Comment (quarckster) [inline cfme/tests/configure/test_access_control.py]: `Tenant_{}' + fauxfactory.gen_alphanumeric()` I believe a typo is here.

Comment (vatsalparekh) [reply cfme/tests/configure/test_access_control.py]: Corrected.

**Comment by dajoRH** [comment]
Lint report for commit 29cf9b81a25688e576b071e23cf5c608c88e975d:

`cfme/tests/configure/test_access_control.py`:
- :bangbang: Line 1041:15: **E702** *multiple statements on one line (semicolon)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/configure/test_access_control.py [hunk 4]
```diff
@@ -1001,3 +1000,45 @@ def test_tenant_quota_input_validate(appliance):
...
+    'seq',
+    [(1, 2), (2, 1)],
+    ids=['child_more_then_parent', 'parent_less_then_child']
+)
+def test_child_tenant_quota_validation(appliance, request, tenants,
+                                       seq, field, value):
+    import pdb; pdb.set_trace()
+    tenants[seq[0]].set_quota(**{'{}_cb'.format(field): True, field: value + seq[0]})
+    with error.handler('Error when saving tenant quota'):
+        tenants[seq[1]].set_quota(**{'{}_cb'.format(field): True, field: value + seq[1]})
```
Comment (anurag03) [inline cfme/tests/configure/test_access_control.py]: @vatsalparekh: Try to use this method: https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/configure/test_default_views_infra.py#L45, Manipulate this function for your use.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by vatsalparekh** [comment]
This PR is stuck with issue in navigation with Tenants in accordion tree, doubt on WT
waiting for response from @mfalesni

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by vatsalparekh** [comment]
Closing this as @anurag03 already merged similar one

**Pull request closed without merging**